### PR TITLE
Simulation Framework Upgrade

### PR DESCRIPTION
**Pull request by ElliotFriedman**
- Add pre and post simulation pcv checks to ensure PCV does not drop after a proposal
- Add post simulation mint and redeems on all PSM's
- Add post simulation to using PCV guard to withdraw all PCV
- Add pre and post simulation to ensure that oracle price does not change after proposal execution on both PSM's

### CODE REVIEW

**Review by k-xo — COMMENTED** [review]
This is sick! Looking really good so far. Maybe for arbitrum support we can just extract the repeated logic into helper functions, then just have an if statement that checks the chain ID to run with either the mainnet addressses or arbitrum addresses -- Like you did in the oracle verification

File: contracts/test/integration/utils/MintRedeemVerification.sol [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+        uint256 endingUserUnderlyingBalance = underlying.balanceOf(
+            address(this)
+        );
+        uint256 endingPSMUnderlyingBalance = underlying.balanceOf(address(psm));
+        uint256 endingPSMVoltBalance = volt.balanceOf(address(psm));
+
+        require(
+            endingPSMUnderlyingBalance ==
+                startingPSMUnderlyingBalance - amountOut,
+            "RedeemVerification: a0"
```
Comment (k-xo) [inline contracts/test/integration/utils/MintRedeemVerification.sol:69]: We could have a clearer error message "RedeemVerification: PSM Underlying Mismatch" & similar for Volt mismatch and the same for the user variations for that

Comment (ElliotFriedman) [reply contracts/test/integration/utils/MintRedeemVerification.sol:69]: agreed we can improve this

**Comment by ElliotFriedman** [comment]
> For arbitrum support we can just have an if statement that checks the chain ID to run with either the mainnet addressses or arbitrum addresses -- Like you did in the oracle verification

Mint redeem verification is completely separate logic on arbitrum because you have to impersonate a different address and call a different function to mint volt as it isn't the standard Volt.sol token we're used to. PCV Guard verification can be though.

File: contracts/test/integration/utils/OracleVerification.sol [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            ? allMainnetPSMs
+            : allArbitrumPSMs;
+        for (uint256 i = 0; i < psms.length; i++) {
+            require(
+                oraclePrices[i] == IOracleRef(psms[i]).readOracle().value,
+                "OracleVerification: Price not the same after proposal"
+            );
+        }
+    }
+}
```
Comment (k-xo) [inline contracts/test/integration/utils/OracleVerification.sol:50]: would we also want to check that they're pointing to the same oracle? I know some proposals would change the oracle being pointed to, but I think those would be few & far between so could be worth considering, & commenting out when necessary? wdyt

Comment (ElliotFriedman) [reply contracts/test/integration/utils/OracleVerification.sol:50]: sure, we can check that they're pointing to the proper oracle

**Review by k-xo — APPROVED** [review]

**Pull request merged**